src/main/java/dev/monogon/cue/lang/navigation/CueStructureViewElement.java

```diff
@@ -0,0 +1,87 @@
+package dev.monogon.cue.lang.navigation;
+
+import com.intellij.ide.structureView.StructureViewTreeElement;
+import com.intellij.ide.util.treeView.smartTree.TreeElement;
+import com.intellij.navigation.ItemPresentation;
+import com.intellij.psi.util.PsiTreeUtil;
+import dev.monogon.cue.lang.psi.*;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Copyright (C) 22.04.23 - REstore NV
+ */
+
+public class CueStructureViewElement implements StructureViewTreeElement {
```

Comment:
Most existing implementations in the SDK extend `PsiTreeElementBase`. Is there a reason not to use it?